opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/NoFieldQuery.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.opensearch.storage.script.filter.lucene.relevance;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.sql.common.antlr.SyntaxCheckException;
+import org.opensearch.sql.exception.SemanticCheckException;
+import org.opensearch.sql.expression.FunctionExpression;
+import org.opensearch.sql.expression.NamedArgumentExpression;
+
+/**
+ * Base class to represent relevance queries that search multiple fields.
+ *
+ * @param <T> The builder class for the OpenSearch query.
+ */
+abstract class NoFieldQuery<T extends QueryBuilder> extends RelevanceQuery<T> {
```

Comment:
So `NoField` actually means search on multiple fields? Then is `MultiField[Base]Query` more clear?

Reply:
I apologize for the language used here. The code was based on `MultiFieldQuery.java`, but the text in the comment was not updated. `NoFieldQuery.java` was added to support functions that do not take a 'fields' array as an argument such as the `query` function which would be formatted as `query("field1 : query1")` rather than something like `query_string` which would use `query_string(["field1"], "query1")`. I will update the comment to be correct.

Reply:
Addressed in https://github.com/opensearch-project/sql/pull/1010/commits/e73cf7dd5f21f51f250857fa71fc29eb3d1894f0